### PR TITLE
opensource neon code

### PR DESCRIPTION
**Pull request by Presburger**
issue: #141

### CODE REVIEW

File: src/simd/hook.cc [hunk 1]
```diff
@@ -124,6 +128,22 @@ fvec_hook(std::string& simd_type) {
         simd_type = "GENERIC";
     }
 #endif
+
+#if defined(__ARM_NEON)
+    fvec_inner_product = fvec_inner_product_neon;
```
Comment (liliu-z) [inline src/simd/hook.cc:133]: Just curios, is there any context why we are using raw functions with different names instead of a class with template?

Comment (Presburger) [reply src/simd/hook.cc:133]: template cannot work on the code runtime, template need choosed on the compiling time.

**Pull request merged**